### PR TITLE
Fed filing error

### PR DESCRIPTION
**Pull request by deetz99**
issue: #226 based on #223 

- add alert category enum
- update business type
- add future effective filing i18n alert text
- throw error and add alert if hasFutureEffectiveFilings = true
- update business store tests

Screenshot 2024-06-21 at 4 03 59 PM

### CODE REVIEW

File: web/site/pages/index.vue [hunk 1]
```diff
@@ -102,7 +102,8 @@ if (import.meta.client) {
           AlertCategory.INVALID_TOKEN,
           AlertCategory.BUSINESS_DETAILS,
           AlertCategory.ACCOUNT_ACCESS,
-          AlertCategory.FILING_IN_PROGRESS
+          AlertCategory.FILING_IN_PROGRESS,
+          AlertCategory.FUTURE_EFFECTIVE_FILINGS
```
Comment (deetz99) [inline web/site/pages/index.vue:106]: show alert if future effective filings exists in the alerts array

File: web/site/interfaces/business.ts
Comment (deetz99) [inline web/site/interfaces/business.ts:1]: update business interface

File: web/site/enums/alert-category.ts [hunk 2]
```diff
@@ -17,5 +17,6 @@ export enum AlertCategory {
   TOS_GET_ERROR = 'tos-get-error',
   TOS_PATCH_ERROR = 'tos-patch-error',
   FILING_IN_PROGRESS = 'filing-in-progress',
-  DOCUMENT_DOWNLOAD = 'document-download'
+  DOCUMENT_DOWNLOAD = 'document-download',
+  FUTURE_EFFECTIVE_FILINGS = 'future-effective-filings'
```
Comment (deetz99) [inline web/site/enums/alert-category.ts:21]: add alert category

File: web/site/locales/en-CA.ts [hunk 3]
```diff
@@ -51,6 +51,10 @@ export default {
     'document-download': {
       title: 'Download Failed',
       description: 'Unable to download the document at this time. Please try again or contact support if the issue persists.'
+    },
+    'future-effective-filings': {
```
Comment (deetz99) [inline web/site/locales/en-CA.ts:55]: alert text to match category

File: web/site/stores/business.ts [hunk 4]
```diff
@@ -48,6 +48,15 @@ export const useBusinessStore = defineStore('bar-sbc-business-store', () => {
   function assignBusinessStoreValues (bus: BusinessFull) {
     currentBusiness.value = bus
 
+    // throw error if business has future effective filings
+    if (bus.hasFutureEffectiveFilings) {
```
Comment (deetz99) [inline web/site/stores/business.ts:52]: throw error and add alert if business object hasFutureEffectiveFilings = true

File: web/site/tests/mocks/mockedData.ts
Comment (deetz99) [inline web/site/tests/mocks/mockedData.ts:1]: cleanup mock data

File: web/site/tests/unit/stores/business.test.ts
Comment (deetz99) [inline web/site/tests/unit/stores/business.test.ts:1]: update business store tests

**Review by thorwolpert — APPROVED** [review]

**Pull request merged**